server/src/main/java/org/elasticsearch/plugins/TracingPlugin.java

```diff
@@ -8,13 +8,21 @@
 
 package org.elasticsearch.plugins;
 
-import org.elasticsearch.tasks.Task;
+import java.util.Map;
 
 public interface TracingPlugin {
 
+    interface Traceable {
+        String getSpanId();
+
+        String getSpanName();
+
+        Map<String, Object> getAttributes();
+    }
+
     interface Tracer {
-        void onTaskRegistered(Task task);
+        void onRegistered(Traceable traceable);
 
-        void onTaskUnregistered(Task task);
+        void onUnregistered(Traceable traceable);
```

Comment:
I believe registered/unregistered is still related to the task naming.
Should generic methods be called `onSpanStarted`/`onStarted` or something similar?

Reply:
Thanks. I changed the method names to `onTraceStarted` and `onTraceStopped`.